packages/patternfly-3/patternfly-react/src/components/DualListSelector/DualListControlled.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import { isEqual } from 'lodash';
+import PropTypes from 'prop-types';
+import { DualList } from './index';
+import { arrangeArray, getFilteredResults, adjustProps } from './helpers';
+import { noop } from '../../common/helpers';
+
+class DualListControlled extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      prevProps: props,
+      ...adjustProps(props)
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (!isEqual(nextProps, prevState.prevProps)) {
+      return {
+        prevProps: nextProps,
+        ...adjustProps(nextProps)
+      };
+    }
+    return {
+      ...prevState
+    };
+  }
+
+  onItemChange = e => {
+    const {
+      checked,
+      attributes: {
+        position: { value: position },
+        side: { value: side }
+      }
+    } = e.target;
```

Comment:
It would be nice to have the logic to determine the changes in the `DualList` and have the callbacks take the changed values as parameters.

Reply:
Great idea! passed all the logic to `DualList` and receive the callbacks in `DualListControlled`